src/emulator/storage/apis/firebase.ts

```diff
@@ -120,7 +120,8 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
         data = gunzipSync(data);
       }
       res.setHeader("Accept-Ranges", "bytes");
-      res.setHeader("Content-Type", metadata.contentType);
+      res.setHeader("Content-Type", metadata.contentType || "application/octet-stream");
+      res.setHeader("Content-Disposition", metadata.contentDisposition || "inline");
```

Comment:
Shouldn't these fields be set already when uploading on object to the bucket?

Reply:
So the way Content-Type works:

*  is if it's not passed in the upload request, the metadata field is empty but when fetching it, it defaults to "application/octet-stream"
* if the file name has a supported file extension e.g. .jpg, I believe the server automatically fills the field with the appropriate content type, e.g. image/jpg
* if you pass in an explicit metadata field, that overrides the above behaviors


This change adds support for the first bullet, the third bullet already works in the emulator. We don't have support for the inferred content type, unclear if we want to prioritize something like that.